### ISSUE TITLE
`Invalid cross-device link` crash when working with temporary files

### ISSUE THREAD

**Issue by kirillt** [ARK-Builders/arklib#70]
Simplified procedure for atomic writing is "write to a temporary file, then hard link to the destination". This becomes a problem when the "destination" and "temporary file" reside on different filesystems, e.g. `/tmp` and `/home`.

Possible solutions:
* Create temporary files in same folder as hidden files, then delete them
* Use plain copy instead of hard link (slower?)

**Comment by kirillt** [comment]
**Workaround:** create the folder for temporary files manually and bind it using `export TMPDIR`.

**Comment by gwendalF** [comment]
@kirillt Do you have an example where it failed? It seems that the tempory files are created at the same root than the destination:
    ```
    pub fn make_temp(&self) -> Result<TmpFile> {
        TmpFile::create_in(&self.directory)
    }
    ```

**Comment by kirillt** [comment]
@gwendalF usually, temporary folder is always `/tmp`, but Shelf keeps the data by default in `~/.ark-shelf`.

I can double-check later. Maybe it also depends on where the binary is.

**Comment by kirillt** [comment]
https://github.com/ARK-Builders/arklib/assets/581023/1e6f9a9a-f488-496c-916a-268a08a4ac33

**Comment by kirillt** [comment]
Solved in #88
